discovery/data_sharing/v0.1/data_sharing.md

```diff
@@ -0,0 +1,297 @@
+  subscription or backfill request.
+* `objectUris` is an array of URIs representing the objects.
+* `eventType` MUST be present for events and be one of `new`, `update`,
+  `delete` or `trending`. The first three are for subscriptions to the
+  `lifecycle` type and the latter for the `trends` type.
+  This key MUST NOT be present for responses to backfill requests.
+* `cursor` MAY optionally be present on responses to backfill requests.
+  If it is string its value can be included in a future backfill request
+  to make sure the fediverse server does not send the same object uris
+  again.  If this is a partial announcement that does not conclude an
+  open backfill request this key MUST be absent.
+* `moreObjectsAvailable` MAY optionally be present on responses to
+  backfill requests. If set to `false` it means that no more objects of
+  this category are available and no future backfill requests should be
+  made.
+
+Requests MUST include at least one object URI, but the fediverse server
+MAY save requests and batch events together. In that case it MUST
+respect the `maxBatchSize` it received with the subscription.
+
+In case of backfill requests the fediverse server can decide to send the
+requested number of objects in a single request or split it up into
+smaller requests to reduce the load. In that case the optional `cursor`
+MUST only be included with the final request, but only if more data
+would be available.
+
+Example payload:
+
+```json
+{
+  "source": {
+    "subscriptionId": 58152
+  },
+  "eventType": "new",
+  "objectType": "Note",
+  "objectUris": [
+    "https://fediverse-server.example.com/@example/2342",
+    "https://fediverse-server.example.com/@other/8726"
+  ]
+}
+```
+
+## Retrieving Content From Its Origin (FASP => Wider Fediverse)
+
+As displayed in the sections above, FASP will only receive object URIs
+from connected fediverse servers. They will then need to fetch the
+actual content themselves.
+
+To do so, they need to send HTTP `GET` requests to the object URIs with
+an `Accept` header with the `application/ld+json;
+profile="https://www.w3.org/ns/activitystreams"` media type as specified
+by [ActivityPub](https://www.w3.org/TR/activitypub/#retrieving-objects).
+
+These requests MUST be signed. To achieve a maximum of compatibility
+with existing fediverse software, FASP MUST support request signing with
+both "HTTP Message Signatures" as defined by
+[RFC-9421](https://tools.ietf.org/html/rfc9421.html) as well as the
+earlier draft version "HTTP Signatures" as defined by
+[cavage-12](https://datatracker.ietf.org/doc/html/draft-cavage-http-signatures).
+
+To find out which version a given fediverse server supports, FASP should
+implement "double-knocking": They should first attempt a request using
+"HTTP Message Signatures" and if the fediverse server replies with an
+HTTP status code of `401` or `403` make a second attempt with the older
+draft version, "HTTP Signatures".
```

Comment:
Why not just always require sending signatures? If the server doesn't need them, it'll discard them. That'll reduce request loads by half.

Reply:
We do require that, we just propose to try the final version before falling back to the draft one.

They use the same HTTP header and thus cannot both be used at the same time. I included a link to the SWICG document about this and this is basically their recommendation I just followed here.

Reply:
I'd probably move the link to the SWICG report up, and perhaps we can do some clarifications to that report, e.g., "How to send requests with signatures" which is kind of mentioned, but saying "follow this process, but do double knocking" (i.e., try latest spec, if that fails, try cavage-12). This is kinda awkwardly placed in that document.